src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -692,4 +692,29 @@ export default class ReimbursementRequestService {
     if (!fileData) throw new NotFoundException('Image File', fileId);
     return fileData;
   }
+
+  /**
+   * Edits the vendor name
+   *
+   * @param name the new vendor name
+   * @param vendorId the requested vendor to be edited
+   * @param submitter the user editing the vendor name
+   * @returns the updated vendor
+   */
+  static async editVendors(name: string, vendorId: string, submitter: User) {
+    if (!isHead(submitter.role)) throw new AccessDeniedAdminOnlyException('only Admins and Heads can edit vendors');
```

Comment:
From the ticket description, only the general admin or finance team head/leads can edit vendors. In other words, we want to check if the submitter.role is at least in an admin role, or if the submitter is in the finance team as a lead or head.